tests/acceptance/17_users/unsafe/20_modify_user_missing_secondary_group.cf

```diff
@@ -0,0 +1,52 @@
+bundle common test_meta
+{
+  vars:
+    "description" string => "Managing a user without specifying any secondary groups attribute should result in no changes to existing secondary groups";
+    "store_id" string => "3710";
+    "covers" string => "operational_repaired";
+}
+
+body common control
+{
+      inputs => { "../../default.cf.sub", "user_queries.cf.sub" };
+      bundlesequence  => { default("$(this.promise_filename)") };
+}
+
+bundle agent init
+{
+  commands:
+    "/usr/bin/sudo $(user_tests.pw_path) userdel testu";
+    "/usr/bin/sudo $(user_tests.pw_path) delgroup testu";
+    "/usr/bin/sudo $(user_tests.pw_path) delgroup testg1";
+    "/usr/bin/sudo $(user_tests.pw_path) delgroup testg2";
+    "/usr/bin/sudo $(user_tests.pw_path) groupadd testg1";
+    "/usr/bin/sudo $(user_tests.pw_path) useradd testu";
+    freebsd::
+      "/usr/bin/sudo $(user_tests.pw_path) usermod testu --groups testg1 --append";
+    !freebsd::
+      "/usr/bin/sudo $(user_tests.pw_path) usermod --groups testg1 --append testu";
+}
+
+bundle agent test
+{
+  users:
+    "testu"
+      policy => "present";
+}
+
+bundle agent check
+{
+  methods:
+    "any" usebundle => user_exists("testu", "success", "failure"),
```

Comment:
Agree. These should really be the same as the other test I added. Fixing.

Reply:
I see a couple oddities here.

First, it's atypical to define metadata in a meta prefixed bundle.

Instead of 

    bundle common test_meta
    {
      vars:
          "description" string => "Managing a user without specifying any secondary groups attribute should result in no changes to existing secondary groups";
          "store_id" string => "3710";
          "covers" string => "operational_repaired";
    }

I would expect:

    bundle agent test
    {
      meta:
          "description" string => "Managing a user without specifying any secondary groups attribute should result in no changes to existing secondary groups";
          "store_id" string => "3710";
          "covers" string => "operational_repaired";
    }

Technically it works either way since *meta* type promises define vars in the `meta_$(this.bundle)` scope.

Second, I think that `store_id` should have been `story_id`.

I personally don't recall every using the `covers` or `story_id` metadata, but I see that there is some [support for them](https://github.com/cfengine/core/blob/c964d056ec5225c0ed0e399d7ebbd2249fe8cc2b/tests/acceptance/testall#L502-L515) in `testall`. Only *description* is [mentioned in the README](https://github.com/cfengine/core/blob/c964d056ec5225c0ed0e399d7ebbd2249fe8cc2b/tests/acceptance/README#L218-L223).